docs/instructlab-cli-1.0.0.md

```diff
@@ -0,0 +1,35 @@
+# The Road to 1.0.0
+
+_Or: How I Learned to Stop Worrying and Love to GA_
+
+## Context and Goals
+
+At the time of this writing, the InstructLab CLI has gone from a scrappy research project to an upstream community serving as the basis for multiple downstreams, with the goal to continuing to evolve the community to encourage more participation from additional stakeholders. To wit, it would behoove us to determine what exactly we should be roadmapping between now and a proper 1.0.0 release, which demonstrates the following to existing and potential community members:
+
+1. An official goalpost for the community denoting the evolution of InstructLab from a pre-1.0 project lacking the stability and supportability typically seen from 1.0-and-beyond projects.
+1. A dedicated set of V1 interfaces that can be counted on for continuous usage of InstructLab 1.0 with future provisions made for backwards compatibility for subsequent Y-Streams and Z-Streams.
+1. A commitment from the Oversight Committee and Maintainer teams to continue to maintain InstructLab throughout a 1.y cycle and work towards an eventual 2.0.
+
+## MVP for an InstructLab CLI 1.0.0
+
+At a high-level, these are the items the Maintainer teams believe should serve as prereqs for releasing an InstructLab CLI 1.0.0:
+
+- Removal of the need for a `config.yaml` with a fully-realized configuration scheme, centered around the usage of system profiles
+- An official v1 REST API schema
+- Integration of the InstructLab CLI with RAG
+- An upgrade path to subsequent Y-Streams and an eventual 2.0
+- Backwards compatibility across the 1.y stream
```

Comment:
IMO this is a really big commitment that we don't necessarily have to make. It's common for consumers of python projects to expect minimal backward compat and to take the most recent version as that one with fewest feature errors.

Reply:
Would a compromise be to commit to backward compatibility across all 1.0.y versions?  I.e., that API breaking changes will not happen on third digit point releases.

Reply:
Good point, the vagueness of this makes it quite the lift - maybe we can commit to backwards compatibility for N-1 Y-Streams?

E.g., `1.4` is backwards-compatible with `1.3` but not _necessarily_ `1.2` 

Or do you think even that might be more than worth committing to?

Reply:
N-1-Y sounds complicated to manage to me because breaking changes need to be staged out across multiple releases.

Reply:
Def good with backwards compatibility across Z-Streams only - @JamesKunstle wdyt?

Reply:
Barring any input, this will go in as-is on Friday - can update it in a followup